File: examples/platforms/wasm/tests/records/with_strings.test.mjs
import { assert, demo } from "../support/index.mjs";

export async function run() {
  // case:records.with_strings.person.basic
  const person = demo.makePerson("Ali", 30);
  assert.deepEqual(demo.echoPerson(person), person);
  assert.equal(demo.greetPerson(person), "Hello, Ali! You are 30 years old.");

  // case:records.with_strings.address.basic
  const address = { street: "Main St", city: "Amsterdam", zip: "1000AA" };
  assert.deepEqual(demo.echoAddress(address), address);
  assert.equal(demo.formatAddress(address), "Main St, Amsterdam, 1000AA");
}

File: examples/platforms/dart/test/records/with_strings_test.dart
import 'package:test/test.dart';
import 'package:demo/demo.dart';

void main() {
  tearDownAll(shutdownBoltffi);
  test('records with strings', () {
    final person = makePerson('Alice', 30);
    expect(
      person,
      isA<Person>()
          .having((p) => p.name, "`Person` name property", equals('Alice'))
          .having((p) => p.age, "`Person` age property", equals(30)),
      reason: 'case:records.with_strings.person.should_make_from_fields',
    );

    expect(
      echoPerson(person),
      person,
      reason: 'case:records.with_strings.person.should_roundtrip_value',
    );

    expect(
      greetPerson(person),
      'Hello, Alice! You are 30 years old.',
      reason: 'case:records.with_strings.person.should_format_greeting',
    );

    final emojiPerson = makePerson('🎉 Party', 25);
    expect(emojiPerson.name, '🎉 Party');

    final echoedEmojiPerson = echoPerson(emojiPerson);
    expect(echoedEmojiPerson.name, '🎉 Party');

    final address = Address(
      street: '123 Main St',
      city: 'Springfield',
      zip: '12345',
    );

    expect(
      echoAddress(address),
      address,
      reason: 'case:records.with_strings.address.should_roundtrip_value',
    );

    expect(
      formatAddress(address),
      '123 Main St, Springfield, 12345',
      reason: 'case:records.with_strings.address.should_format_value',
    );
  });
}
